zerver/actions/realm_settings.py

```diff
@@ -297,6 +298,10 @@ def do_deactivate_realm(realm: Realm, *, acting_user: Optional[UserProfile]) ->
 
 
 def do_reactivate_realm(realm: Realm) -> None:
+    if not realm.deactivated:
+        logging.warning("do_reactivate_realm run on realm %s, which is already active.", realm.id)
```

Comment:
I think just "Realm %s cannot be reactivated because is already active." would be a bit more friendly phrasing.

Otherwise this lgtm (don't know why it failed CI, but please get it passing and tag for integration review).